### PR TITLE
Refactor rocoto task XML creation

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
# Description
Refactors the rocoto task generation to be recursive. This will allow nested metatasks to loop over multiple variables, which is needed for GEFS product generation.

As part of this refactor, there is no longer separate arguments to designate metatasks. Instead, task dicts can include a nested 'task_dict' as well as a 'var_dict' containing the variables to loop over. The nested task dict can then either have another layer, or be the innermost task.

To accommodate the new recursive nature, some defaults that were previously defined in `create_wf_task()` had to be pushed down into the function that creates the innermost task. Also, former keywords have been absorbed by the task dict.

Refs #823
Refs #827

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
- Cycled test on Orion
- Forecast-only test on Orion
- GEFS test on Orion

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
Not quite working yet, still need to make some tweaks.

**Comment by WalterKolczynski-NOAA** [comment]
Okay, think I have the bug worked out.

**Review by HenryRWinterbottom — COMMENTED** [review]
This LGTM. The docstrings are helpful and making the dictionaries more readable is easy on the eyes. This is beyond the scope of this PR, but with some time investment I think some of the redundant code blocks can be parsed out and stuffed in wrapper functions (e.g., decorators) to make this code even more readable. But, just my $0.02.

**Comment by WalterKolczynski-NOAA** [comment]
Orion tests are randomly failing due to stalling on module load.

File: workflow/rocoto/rocoto.py [hunk 1]
```diff
@@ -76,7 +123,7 @@ def create_task(task_dict: Dict[str, Any]) -> List[str]:
     nodes = resources_dict.get('nodes', 1)
     ppn = resources_dict.get('ppn', 1)
     threads = resources_dict.get('threads', 1)
-    log = task_dict.get('log', 'demo.log')
+    log = task_dict.get('log', f'&ROTDIR;/logs/@Y@m@d@H/{taskname}.log')
```
Comment (aerorahul) [inline workflow/rocoto/rocoto.py]: I think this needs to be reverted.

**Review by aerorahul — COMMENTED** [review]
mostly looks good.  I think one tiny thing slipped through.

File: workflow/rocoto/rocoto.py (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply workflow/rocoto/rocoto.py]: Oops. Fixed.

**Review by aerorahul — APPROVED** [review]

**Pull request merged**